internal/exec/runner.go

```diff
@@ -179,14 +181,45 @@ func (r *runner) Files(args []string) error {
 	return nil
 }
 
-func (r *runner) Compile(args []string, dryRun bool) error {
+func (r *runner) Compile(args []string, dryRun bool, outputPath string) error {
 	meta, err := r.getMeta(args, 1)
 	if err != nil {
 		return err
 	}
 	r.printAffectedFiles(meta)
-	_, err = r.compile(false, false, dryRun, meta)
-	return err
+
+	if outputPath == "" {
+		// validate the protobuf files and return
+		_, err = r.compile(false, false, dryRun, meta)
+		return err
+	}
+
+	// otherwise, dump FileDescriptorSets to outputPath
+	fileDescriptorSets, err := r.compile(false, true, dryRun, meta)
+	if err != nil {
+		return err
+	}
+	return r.fileDescriptorSetsToFile(outputPath, fileDescriptorSets)
+}
+
+// writes a single FileDescriptorSet containing all given FileDescriptorProtos
+func (r *runner) fileDescriptorSetsToFile(outputPath string, sets []*descriptor.FileDescriptorSet) error {
+	// TODO(peats-bond): have an option (-w) to overwrite output file
+	file := filepath.Join(r.workDirPath, outputPath)
+	if _, err := os.Stat(file); err == nil {
+		return fmt.Errorf("%s already exists", file)
+	}
+
+	if err := os.MkdirAll(filepath.Dir(outputPath), 0755 /*user:rwx group:rx other: rx*/); err != nil {
```

Comment:
nit: everyone knows what 0755 is (or should lol)

Reply:
xD